model/src/main/java/de/danoeh/antennapod/model/feed/Transcript.java

```diff
@@ -1,23 +1,69 @@
 package de.danoeh.antennapod.model.feed;
 
+import java.util.Hashtable;
 import java.util.Map;
+import java.util.Set;
 import java.util.TreeMap;
 
 public class Transcript {
 
     private final TreeMap<Long, TranscriptSegment> segmentsMap = new TreeMap<>();
+    private Set<String> speakers;
+    private Hashtable<Long, Integer> positions;
+    private Object[] objs; // array of TreeMap<Long, TranscriptSegment>
```

Comment:
Having TreeMap and Hash table and array somehow feels a bit duplicated. Just a simple array should have the same performance and would be a lot easier to comprehend when reading this code later